simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+                    print("atom type:"+values[0])
+                    #self.dc(CHEMICAL_SPECIE = values[0])
+                    SP[CUBA.CHEMICAL_SPECIE].append(values[0])
+                    mass = float(values[1])
+                    SP[CUBA.MASS].append(mass)
+                    potential_file = values[2]
+                    SP[CUBA.FRICTION_COEFFICIENT].append(potential_file)
+            line = f.next()
+        return line
+
+    def process_k_points(self,f,SP,mode='automatic'):
+        #skip line
+        print('processing k_points section')
+        line = f.next()
+    #    SP[CUBAExtension.K_POINTS_MODE] = mode
+        i = 0
+
+        SP[CUBA.PROBABILITY_COEFFICIENT] = mode
+
+        while _ReadState.get_state(_ReadState.K_POINTS,line) == _ReadState.K_POINTS:
+    #        print('line:'+str(line))
+            values = line.split()
+            if len(values):
+                K_points = (values)
+                print('k points:'+str(K_points))
+                SP[CUBA.EQUATION_OF_STATE_COEFFICIENT] = K_points
+
+            line = f.next()
+        return line
+
+
+    #I am not sure whether to use datacontainer or particlecontainer - maybe PC goes in DC?
+    def process_atomic_positions(self,f,pc,SP,units='(angstrom)'):
+        print('processing atomic_positions section')
+        try:
+            line = f.next()
+            SP[CUBA.KINEMATIC_VISCOSITY] = units
+            particle_list = []
+            while _ReadState.get_state(_ReadState.ATOMIC_POSITIONS,line) == _ReadState.ATOMIC_POSITIONS:
+                logging.debug('line in atomic positions section:'+str(line))
+                values = line.split()
+                print('values:'+str(values))
+                atom_pos = [0,0,0]
+                i = 0
+                if values[0] in atomtypes:
+                    atomtype = values[0]
+                    atom_id = i
+                    atom_pos[0] = float(values[1]) * 1e-10  #store position in meters; original in Angstrom
+                    atom_pos[1] = float(values[2]) * 1e-10
+                    atom_pos[2] = float(values[3]) * 1e-10
+                    s = str(i)
+                    #make uid using string of index
+                    #uidstring =
+                    #uid = uuid.UUID(uidstring)
+
+                    p = Particle([atom_pos[0],atom_pos[1],atom_pos[2]])  #,uuid.UUID(int=i)
+                    print('uid:'+str(p.uid))
+                    p.data[CUBA.CHEMICAL_SPECIE] = atomtype
+                    particle_list.append(p)
+                    try:
+                        line = f.next()
+                    except StopIteration:
+                        print('EOF')
+                        break
+            if len(particle_list):
+                pc.add_particles(particle_list)
+                  #  part_container.add_particle(p)
+                n=self.count_particles()
+                print('n_particles='+str(n))
+        except StopIteration:
+            return('EOF')
```

Comment:
The usage of `process_atomic_position` looks like it always expects a PC, but this statement here does something else.   Not certain what the goal of this statement is though.  Quickly glancing at it, it looks like you skip the first line of atom positions section and the `StopIteration` caught here would occur if there were no lines of the atom positions...